pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,390 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/route/api"
+	fileutil "github.com/openshift/origin/pkg/util/file"
+)
+
+const (
+	routeLong = `
+Expose containers externally via secured routes
+
+The name of the service you want to expose needs to be specified by using the
+--expose option.
+
+Three types of secured routes are supported: edge, passthrough, and reencrypt.
+If you wish to create unsecured routes, see "%[1]s expose -h"
+`
+)
+
+// NewCmdCreateRoute is a macro command to create a secured route.
+func NewCmdCreateRoute(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "route",
+		Short: "Expose containers externally via secured routes",
+		Long:  fmt.Sprintf(routeLong, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Help()
+		},
+	}
+
+	cmd.PersistentFlags().String("hostname", "", "Set a hostname for the new route")
+	cmd.PersistentFlags().String("port", "", "Name of the service port or number of the container port the route will route traffic to")
+	cmd.PersistentFlags().String("expose", "", "Name of the service that the new route is exposing")
```

Comment:
So I really am sorry about this.  Turns out that `oc create route edge --help` doesn't show me the `expose` flag.